pkg/splunk/enterprise/clustermanager.go

```diff
@@ -434,3 +451,160 @@ func VerifyCMisMultisite(ctx context.Context, cr *enterpriseApi.ClusterManager,
 	}
 	return extraEnv, err
 }
+
+// upgradeScenario checks if it is suitable to update the clusterManager based on the Status of the licenseManager, returns bool, err accordingly
+func upgradeScenario(ctx context.Context, c splcommon.ControllerClient, cr *enterpriseApi.ClusterManager) (bool, error) {
```

Comment:
nit:maybe change the name of the function to be something like `isReadyForUpgrade`?